benches/benches/README.md

```diff
@@ -0,0 +1,194 @@
+# SV1 and SV2 clients benchmarks
+This directory contains code that benchmarks the performance of sv1 and sv2 clients using the `criterion` and `iai` library.
+
+The benchmarking project aims to compare the performance of sv2 against sv1 and demonstrate the improvements of sv2. The benchmarks measure various aspects of the clients' performance, including  subscription latency, share submission time, systems requirements such as RAM access, Instructions per cycle and more..
+
+## Project Structure
+The project is structured as follows:
+
+- The benches/ directory contains the main project files.
+- The src/ directory contains subdirectories for sv1 and sv2 clients.
+- Each client directory contains benchmark files criterion_svX_benchmark.rs and iai_svX_benchmark.rs for sv1 and sv2 respectively.
+- The lib/client.rs file within each client directory contains additional code relating to each client implementation.
+
+
+- criterion_svX_benchmark.rs: Uses the criterion crate for latency benchmarking.
+- iai_benchmarks.rs: Uses the iai crate to measure system requirements and performance.
+
+## Running Benchmarks
+- To run the benchmarks, follow these steps:
+
+- Install Rust and Cargo if you haven't already.
+- Clone this repository
+```sh
+git clone https://github.com/stratum-mining/stratum
+```
+- Navigate to the benches directory.
+```sh
+cd benches
+```
+- Open a terminal and run the following command to execute the benchmarks:
+```sh
+cargo bench
+```
+
+The benchmark results will be displayed in the terminal. `target/criterion` and `target/iai` will also be created, which contains more detailed results.
+
+## Benchmarking
+It is important to note that for now, sv2 does not count encryption and decryption.
+
+The following benchmark functions are available:
+
+### sv1
+1. **Subscription Benchmarks**:
+   - `client_sv1_get_subscribe`: Measures the latency and system requirements of a subscription request.
+   - `client_sv1_subscribe_serialize`: Measures the latency and system requirements it takes to serialize a subscription message.
+   - `client_sv1_serialize_deserialize`: Measures the latency and system requirements it takes to serialize and then deserialize a subscription message.
```

Comment:
`client_sv1_serialize_deserialize` needs to be changed in `client_sv1_subscribe_serialize_deserialize`